UI-5-added-color-sugar-filters

Vlad-9 opened the pull request.

Vlad-9 committed:
UI-5-added-color-sugar-filters

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterItemView.swift:
```diff
@@ -0,0 +1,28 @@
+
+    var body: some View {
+        VStack(alignment: .trailing) {
+            HStack(alignment: .center) { Image(systemName: self.isChecked ? "checkmark.square.fill": "square").foregroundColor(self.isChecked ? Color(red: 158 / 255.0, green: 51 / 255.0, blue: 75 / 255.0) : .black) ; Text(item.title)  .foregroundColor(.primary).font(.system(size: 13)) }
```
Страшилка однострочная

BobCatC replied:
Страшна

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterItemView.swift:
```diff
@@ -0,0 +1,30 @@
+
+    var body: some View {
+        VStack(alignment: .trailing) {
+            HStack(alignment: .center) { Image(systemName: self.isChecked ? "checkmark.square.fill": "square").foregroundColor(self.isChecked ? Color(red: 158 / 255.0, green: 51 / 255.0, blue: 75 / 255.0) : .black) ; Text(item.title).foregroundColor(.primary)
```
опять страшилка

BobCatC replied:
Совершенно не стоит реализовывать полностью совпадающий UI. Гораздо лучше было бы вместо этих мучений с `Image`, обёрнутым в `Button`, использовать `Toggle` (switch по-русски). Его можно будет и забиндить, и убрать этот `onItemTap` (который должен использоваться как раз с кнопкой, а не с switch'ом)

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+import SwiftUI
+
+struct ColorFilterItemButton: View {
```
Копипаста с SugarFilterItemButton

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterItemView.swift:
```diff
@@ -0,0 +1,30 @@
+struct SugarFilterItemView: View {
+
+    let item: SugarFilterItemModel
+    var isChecked: Bool
```
можно сделать @Binding

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+
+    var body: some View {
+        Button(action: action, label: {
+            SugarFilterItemView(item: item, isChecked: self.isChecked).padding()
```
а тут isChecked: $isChecked

dimonker replied:
self лишнее

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterView.swift:
```diff
@@ -0,0 +1,37 @@
+    }
+    
+    typealias OnItemTap = (ColorFilterItemModel) -> Void
+    private func itemDidTap(_ item: ColorFilterItemModel) {
```
Эта функция не используется

dimonker commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterView.swift:
```diff
@@ -0,0 +1,37 @@
+
+struct ColorFilterView: View {
+    var items: [ColorFilterItemModel]
+    var onItemTap: OnItemTap?
```
действия по идее должны быть во ViewModel

BobCatC replied:
Видимо, имелось в виду, что инстанс, который создаёт `ColorFilterView` передаёт этой вьюхе в конструкторе лямбда функцию, чтобы реагировать на изменения. По-аналогии работает Button (у кнопки также есть callback function поле `action`)
@dimonker, или в чём-то другом вопрос?

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+        })
+    }
+
+#if DEBUG
```
Как так получилось, что `ColorFilterItemButtonPreview` оказался внутри ColorFilterItemButton?)

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterView.swift:
```diff
@@ -0,0 +1,37 @@
+    var items: [ColorFilterItemModel]
+    var onItemTap: OnItemTap?
+    var body: some View {
+        VStack {
```
Вложенный `VStack` точно нужен?

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+    }
+
+#if DEBUG
+struct SugarFilterItemButtonPreview: PreviewProvider {
```
Аналогично неясно, как этот тут оказалось

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+    private static let item = SugarFilterItemModel( title: "test")
+    static var previews: some View {
+        return SugarFilterItemButton(item: item)
+
```
Перевод строки?)

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/SugarFilter/SugarFilterView.swift:
```diff
@@ -0,0 +1,38 @@
+                ForEach(items) { item in
+                    SugarFilterItemButton(item: item).frame(minWidth: 60, maxWidth: .infinity)
+                    Divider()
+                    
```
Перевод строки?)

BobCatC commented on WineUp/WineUp/Modules/Catalog/Catalog/Filters/ColorFilter/ColorFilterItemButton.swift:
```diff
@@ -0,0 +1,30 @@
+    private static let item = ColorFilterItemModel( title: "test")
+    static var previews: some View {
+        return ColorFilterItemButton(item: item)
+
```
Перевод строки?)

BobCatC reviewed (changes requested):
1. Нет заголовков файлов (имя файла, кто и когда его создал)
2. Файлы SuperFilterView и ColorFilterView (аналогично их subview) - просто копии друг друга, нужна более обобщённая реализация
3. Есть мелкие замечания по коду

Насчёт неиспользуемых функций и `Toggle` VS `Button`, сейчас взаимодействия всё равно нет, нужен только UI, на это можно забить, а в некст спринте я могу этот момент поправить (уже есть идея, как правильно связать данные через `State` и `Binding`)

Vlad-9 committed:
UI-5 Bug fixes

1) Добавлены заголовки файлов
2) UI реализация фильтров по цвету и количеству сахара объединена в один
3) Исправлены ошибки в коде
